packages/web-components/src/utils/date-utils.ts

```diff
@@ -284,9 +282,12 @@ export function getErrorParameters(
   }
 }
 
-const daysForSelectedMonth = (leapYear: boolean, month: number): number => {
-  return leapYear && month == 2 ? 29 : days[month]?.length || 0;
-}
+// Get last day of the month (month is zero based, so we're +1 month, day 0);
+// new Date() will recalculate and go back to last day of the previous month.
+// Return 31 for undefined month or year to not invalidate the day with
+// partial data (this used to be set to zero by default)
+export const daysForSelectedMonth = (year: number, month: number): number =>
+  year && month ? new Date(year, month, 0).getDate() : 31;
```

Comment:
Also, with a default of `31`, this weird error message doesn't show up - reproduce by entering a number > 12 in month and a number > 31 in day, then blur; clear month (or maybe day?) and this error should show up
<img width="529" alt="empty-1-and-0" src="https://github.com/department-of-veterans-affairs/component-library/assets/136959/dbab783c-448d-4046-8a51-85b9b215b3df">
<img width="527" alt="filled-1-and-0" src="https://github.com/department-of-veterans-affairs/component-library/assets/136959/7aa5206f-20e9-4f18-b73a-fa55116e45c6">